core/src/main/java/hudson/PluginManager.java

```diff
@@ -1344,6 +1346,120 @@ public void stop() {
         LogFactory.release(uberClassLoader);
     }
 
+    @Restricted(NoExternalUse.class)
+    public static boolean isNonMetaLabel(String label) {
+        return !("adopt-this-plugin".equals(label) || "deprecated".equals(label));
+    }
+    
+    @Restricted(NoExternalUse.class)
+    public HttpResponse doPluginsSearch(@QueryParameter String query, @QueryParameter Integer limit) {
+        List<JSONObject> plugins = new ArrayList<>();
```

Comment:
missing permission check?
```suggestion
        Jenkins.get().checkPermission(Jenkins.SYSTEM_READ);
        List<JSONObject> plugins = new ArrayList<>();
```

Reply:
I think already covered here:

https://github.com/timja/jenkins/blob/plugin-manager/core/src/main/java/hudson/PluginManager.java#L2393

the other permission checks are for tightening it to ADMINISTER mostly

Reply:
Tested, `reader is missing the Overall/SystemRead permission`